src/overlays/actors/ovl_En_Bigpamet/z_en_bigpamet.c

```diff
@@ -90,121 +113,763 @@ static DamageTable D_80A29734 = {
+
+        ptr->unk_20 = Rand_ZeroFloat(0.0025000002f) + 0.002f;
+    }
+
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_B_PAMET_BREAK);
+}
+#else
+void func_80A27B58(EnBigpamet* this);
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A27B58.s")
+#endif
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A27DD8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A27FE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A2811C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A281B4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A281DC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A28274.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A282C8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A28378.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A283A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A283F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A2844C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A284E4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A2855C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A28618.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A2866C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A286C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A28708.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A28760.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigpamet/func_80A287E8.s")
+void func_80A27DD8(EnBigpamet* this, PlayState* play) {
+    s32 i;
+    Vec3f sp98;
+    Vec3f sp8C;
+    f32 temp_fs0;
+    f32 temp_fs4;
+    f32 temp_fs5;
+    s16 temp_s0;
+
+    temp_fs4 = Math_SinS(this->actor.wallYaw + 0x4000);
+    temp_fs5 = Math_CosS(this->actor.wallYaw + 0x4000);
+
+    sp8C.x = Math_SinS(this->actor.wallYaw + 0x8000) * 50.0f + this->actor.world.pos.x;
+    sp8C.y = this->actor.world.pos.y;
+    sp8C.z = (Math_CosS(this->actor.wallYaw + 0x8000) * 50.0f) + this->actor.world.pos.z;
+
+    sp8C.x += ((this->actor.world.pos.x - sp8C.x) * 0.3f);
+    sp8C.z += ((this->actor.world.pos.z - sp8C.z) * 0.3f);
+
+    for (i = 0; i < 4; i++) {
+        temp_fs0 = Rand_ZeroFloat(60.0f) + 50.0f;
+        temp_s0 = Rand_Next() >> 0x11;
+
+        sp98.x = (Math_CosS(temp_s0) * temp_fs0 * temp_fs4) + sp8C.x;
+        sp98.y = (Math_SinS(temp_s0) * temp_fs0) + sp8C.y;
+        sp98.z = (Math_CosS(temp_s0) * temp_fs0 * temp_fs5) + sp8C.z;
+
+        func_800B12F0(play, &sp98, &gZeroVec3f, &gZeroVec3f, Rand_S16Offset(950, 0x64), Rand_S16Offset(0x14, 0xA), 20);
```

Comment:
```suggestion
    Vec3f pos;
    Vec3f sp8C;
    f32 temp_fs0;
    f32 temp_fs4;
    f32 temp_fs5;
    s16 temp_s0;

    temp_fs4 = Math_SinS(this->actor.wallYaw + 0x4000);
    temp_fs5 = Math_CosS(this->actor.wallYaw + 0x4000);

    sp8C.x = Math_SinS(this->actor.wallYaw + 0x8000) * 50.0f + this->actor.world.pos.x;
    sp8C.y = this->actor.world.pos.y;
    sp8C.z = (Math_CosS(this->actor.wallYaw + 0x8000) * 50.0f) + this->actor.world.pos.z;

    sp8C.x += ((this->actor.world.pos.x - sp8C.x) * 0.3f);
    sp8C.z += ((this->actor.world.pos.z - sp8C.z) * 0.3f);

    for (i = 0; i < 4; i++) {
        temp_fs0 = Rand_ZeroFloat(60.0f) + 50.0f;
        temp_s0 = Rand_Next() >> 0x11;

        pos.x = (Math_CosS(temp_s0) * temp_fs0 * temp_fs4) + sp8C.x;
        pos.y = (Math_SinS(temp_s0) * temp_fs0) + sp8C.y;
        pos.z = (Math_CosS(temp_s0) * temp_fs0 * temp_fs5) + sp8C.z;

        func_800B12F0(play, &pos, &gZeroVec3f, &gZeroVec3f, Rand_S16Offset(950, 0x64), Rand_S16Offset(0x14, 0xA), 20);
```
Minor Documentation request